packages/api/lib/writeRecords/write-granules.js

```diff
@@ -462,34 +491,59 @@ const _writeGranuleRecords = async (params) => {
     writeConstraints = true,
   } = params;
   let pgGranule;
+  let writePgGranuleResult;
 
   log.info('About to write granule record %j to PostgreSQL', postgresGranuleRecord);
   log.info('About to write granule record %j to DynamoDB', apiGranuleRecord);
 
   try {
     await createRejectableTransaction(knex, async (trx) => {
-      pgGranule = await _writePostgresGranuleViaTransaction({
+      writePgGranuleResult = await _writePostgresGranuleViaTransaction({
         granuleRecord: postgresGranuleRecord,
         executionCumulusId,
         trx,
         granulePgModel,
         writeConstraints,
       });
-
-      // Future: refactor to cover the entire object?
-      // Ensure PG default createdAt value is propagated to DynamoDB/ES
-      // in the case where _writeGranule is called without createdAt set
-      if (!apiGranuleRecord.createdAt) {
-        apiGranuleRecord.createdAt = pgGranule.created_at.getTime();
+      if (writePgGranuleResult.status === 'dropped') {
+        return;
       }
+      pgGranule = writePgGranuleResult.pgGranule;
 
-      await granuleModel.storeGranule(apiGranuleRecord, writeConstraints);
-      await upsertGranule({
-        esClient,
-        updates: apiGranuleRecord,
-        index: process.env.ES_INDEX,
-      }, writeConstraints);
+      if (writeConstraints && isStatusActiveState(pgGranule.status)) {
+        // pgGranule was updated, but with writeConstraints conditions, so only some values were
+        // updated. we need to ensure the correct values are propogated to Dynamo and ES
+        const limitedUpdateApiGranuleRecord = await translatePostgresGranuleToApiGranule({
+          granulePgRecord: pgGranule,
+          knexOrTransaction: trx,
+        });
+
+        await granuleModel.storeGranule(limitedUpdateApiGranuleRecord, writeConstraints);
+        await upsertGranule({
+          esClient,
+          updates: limitedUpdateApiGranuleRecord,
+          index: process.env.ES_INDEX,
+        }, writeConstraints);
+      } else {
+        // Future: refactor to cover the entire object?
+        // Ensure PG default createdAt value is propagated to DynamoDB/ES
+        // in the case where _writeGranule is called without createdAt set
+        if (!apiGranuleRecord.createdAt) {
+          apiGranuleRecord.createdAt = pgGranule.created_at.getTime();
+        }
+
+        // TODO: refactor to not need apiGranuleRecord, only need files and a few other fields
+        await granuleModel.storeGranule(apiGranuleRecord, writeConstraints);
+        await upsertGranule({
+          esClient,
+          updates: apiGranuleRecord,
+          index: process.env.ES_INDEX,
+        }, writeConstraints);
+      }
     });
+    if (writePgGranuleResult.status === 'dropped') {
```

Comment:
This may have a typing concern, depending on the resolution re: the _writePostgresGranuleViaTransaction typing